sw/host/opentitanlib/src/transport/cw310/mod.rs

```diff
@@ -60,13 +64,36 @@ impl CW310 {
     }
 
     // Initialize the IO direction of some basic pins on the board.
-    fn init_direction(&self) -> Result<()> {
+    fn init_direction(&self) -> anyhow::Result<()> {
         let device = self.device.borrow();
         device.pin_set_output(Self::PIN_SRST, true)?;
         device.pin_set_output(Self::PIN_JTAG, true)?;
         device.pin_set_output(Self::PIN_BOOTSTRAP, true)?;
         Ok(())
     }
+
+    fn open_uart(&self, instance: u32) -> Result<SerialPortUart> {
+        let usb = self.device.borrow();
+        let serial_number = usb.get_serial_number();
+
+        let mut ports = serialport::available_ports().wrap(UartError::EnumerationError)?;
+        ports.retain(|port| {
+            if let SerialPortType::UsbPort(info) = &port.port_type {
+                if info.serial_number.as_deref() == Some(serial_number) {
+                    return true;
+                }
+            }
+            false
+        });
+        // The CW board seems to have the last port connected as OpenTitan UART 0.
+        // Reverse the sort order so the last port will be instance 0.
+        ports.sort_by(|a, b| b.port_name.cmp(&a.port_name));
```

Comment:
Can this be `sort_by_key`?

Reply:
This piece of code (all of open_uart) was hoisted from the constructor in cw310/uart.rs in order to make that helper reusable.  I am not familiar with this CW310 debugger, and would rather not modify the code too much.

Reply:
This should be a safe refactor. See: https://doc.rust-lang.org/src/alloc/slice.rs.html#368-371